okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java

```diff
@@ -54,38 +55,27 @@
 /**
  * Client stream for the okhttp transport.
  */
-class OkHttpClientStream extends Http2ClientStream {
+class OkHttpClientStream extends AbstractClientStream2 {
 
   private static final int WINDOW_UPDATE_THRESHOLD = Utils.DEFAULT_WINDOW_SIZE / 2;
 
   private static final Buffer EMPTY_BUFFER = new Buffer();
 
-  @GuardedBy("lock")
-  private int window = Utils.DEFAULT_WINDOW_SIZE;
-  @GuardedBy("lock")
-  private int processedWindow = Utils.DEFAULT_WINDOW_SIZE;
+  public static final int ABSENT_ID = -1;
+
   private final MethodDescriptor<?, ?> method;
   /** {@code null} iff start has been called. */
   private Metadata headers;
   private final AsyncFrameWriter frameWriter;
   private final OutboundFlowController outboundFlow;
   private final OkHttpClientTransport transport;
```

Comment:
This looks like it would be good to move to transport state as well. Having a method in transport state would avoid mutating requestHeaders from TransportState as well.